base/XGamepad.ts

```diff
@@ -0,0 +1,503 @@
+import {GoldenSun} from "./GoldenSun";
+
+// export type GamepadState = {
+// 	A: number = 1 << 0;
+// 	B: number = 1 << 1;
+// 	L: number = 1 << 2;
+// 	R: number = 1 << 3;
+
+// 	START: number = 1 << 4;
+// 	SELECT: number = 1 << 5;
+
+// 	LEFT: number = 1 << 6;
+// 	RIGHT: number = 1 << 7;
+// 	UP: number = 1 << 8;
+// 	DOWN: number = 1 << 9;
+// };
+
+/** GBA buttons */
+export enum Button {
+    A,
+    B,
+    L,
+    R,
+    START,
+    SELECT,
+    LEFT,
+    RIGHT,
+    UP,
+    DOWN,
+}
+
+/** GBA button list, for mapping */
+const Buttons = ["A", "B", "L", "R", "START", "SELECT", "LEFT", "RIGHT", "UP", "DOWN"];
+
+/** Custom buttons */
+export enum CButton {
```

Comment:
If this is the list of the buttons of the engine, why do we have LB, RB, LS, and others?

Reply:
XGamepad still needs to know buttons like LT or RT or LS to map the hardware buttons. It also allows user input config to contains these names, ie: `PSY4: LT+Y`.